packages/expo-router/src/fork/StackRouter.tsx

```diff
@@ -0,0 +1,156 @@
+            } else if (action.payload.pop) {
+              route = state.routes.findLast((route) => route.name === action.payload.name);
+            }
+          }
+
+          if (!route) {
+            route = state.preloadedRoutes.find(
+              (route) =>
+                route.name === action.payload.name && id === getId?.({ params: route.params })
+            );
+          }
+
+          let params;
+
+          if (action.type === 'NAVIGATE' && action.payload.merge && route) {
+            params =
+              action.payload.params !== undefined ||
+              routeParamList[action.payload.name] !== undefined
+                ? {
+                    ...routeParamList[action.payload.name],
+                    ...route.params,
+                    ...action.payload.params,
+                  }
+                : route.params;
+          } else {
+            params =
+              routeParamList[action.payload.name] !== undefined
+                ? {
+                    ...routeParamList[action.payload.name],
+                    ...action.payload.params,
+                  }
+                : action.payload.params;
+          }
+
+          let routes: Route<string>[];
+
+          if (route) {
+            if (action.type === 'NAVIGATE' && action.payload.pop) {
+              routes = [];
+
+              // Get all routes until the matching one
+              for (const r of state.routes) {
+                if (r.key === route.key) {
+                  routes.push({
+                    ...route,
+                    path: action.payload.path !== undefined ? action.payload.path : route.path,
+                    params,
+                  });
+                  break;
+                }
+
+                routes.push(r);
+              }
+            } else {
+              /**
+               * React Navigation causes navigation events to behave differently when 'getID()' is used.
+               *
+               * Default behavior (without 'getID()'):
+               *  - 'navigate': If the route is the current route, the params are updated; else the new route is pushed
+               *                onto the stack.
+               *  - 'push': The new route is pushed onto the stack.
+               *
+               * getID() behavior:
+               *  - 'navigate':
+               *       If the route is the current route, the params are updated;
+               *       Else if the route is within the history, the history is rearranged to make the route the current route with
+               *            the new params;
+               *       Else the screen is pushed
+               *  - 'push': Same as `navigate`
+               *
+               *
+               * The purpose of this fork is to maintain the default behavior of navigation events when 'getID()' is used.
+               */
+
+              // START FORK
+              // routes = state.routes.filter((r) => r.key !== route!.key);
```

Comment:
Would be helpful to include a comment with a link to the discussion so it's clear when we can consider reverting this.